Core/Core/AppEnvironment/AppEnvironment.swift

```diff
@@ -99,11 +99,25 @@ open class AppEnvironment {
     }
 
     public var topViewController: UIViewController? {
-        var controller = window?.rootViewController
-        while controller?.presentedViewController != nil {
-            controller = controller?.presentedViewController
+        let locateTopViewController: () -> UIViewController? = {
+            var controller = self.window?.rootViewController
+            while controller?.presentedViewController != nil {
+                controller = controller?.presentedViewController
+            }
+            return controller
+        }
+
+        if Thread.isMainThread {
```

Comment:
couldn't we just trade this whole block with:
`return DispatchQueue.main.sync {
    locateTopViewController()
}`

Reply:
Good idea. I didn't know there was such an overload.